harness/determined/_core/_searcher.py

```diff
@@ -43,9 +57,22 @@ def __init__(
 
     @property
     def length(self) -> int:
+        """
+        length represents the total amount of training which should be reached by the train step
+        before the validate-report steps.
+        """
         return self._length
 
     def report_progress(self, length: float) -> None:
+        """
+        report_progress() tells the master how far along training is, so that the webui can show
```

Comment:
A few issues ...
Change to: "report_progress() indicates master progress, so  the WebUI can show accurate progress reports."
Maybe use "display" instead of "show".

Reply:
"indicates" isn't the right verb, because an indicator is something you check periodically.  The right verb is really "reports".

What about:

```
report_progress() reports the training progress to the Determined master so the WebUI can
show accurate progress to users.
```
?